dbos/queue.go

```diff
@@ -111,17 +111,14 @@ func queueRunner(ctx context.Context) {
 
 	pollingInterval := baseInterval
 
-	// XXX doing this lets the dequeue and the task invokation survive the context cancellation
-	// We might be OK with not doing this. During the tests it results in all sorts of error inside the two functions above due to context cancellation
-	runnerContext := context.WithoutCancel(ctx)
```

Comment:
This causes a small race condition in the TestWorkerConcurrency test -- I might have to change this test altogether eventually